src/Parser/TemplateParser.cs

```diff
@@ -138,22 +139,90 @@ private string ParseName(out int parameterIndex)
                 c = Peek();
                 if (c == '}' || c == ':' || c == ',')
                     parameterIndex = parsed;
-                _position = start;        
+                _position = start;
             }
 
             if (parameterIndex == -1)
                 _isPositional = false;
 
-            return ReadUntil(HoleDelimiters);            
+            return ReadUntil(HoleDelimiters);
         }
 
+        /// <summary>
+        /// Parse format after hole name/index. Handle the escaped { and } in the format. Don't read the last }
+        /// </summary>
+        /// <returns></returns>
         private string ParseFormat()
         {
+
             Skip(':');
-            // TODO: Escaped }} in formats?
-            return ReadUntil('}');
+            string format = null;
+            while (true)
+            {
+                if (format == null)
+                {
+                    //avoid string composition.
+                    format = ReadUntil(TextDelimiters);
+                }
+                else
+                {
+                    format += ReadUntil(TextDelimiters);
+                }
+
+                var c = Read();
+
+                switch (c)
+                {
+                    case '}':
+                        {
+                            if (_position == _length)
+                            {
+                                //unread the }
+                                _position--;
+                                //end of template so done
+                                return format;
```

Comment:
I don't quite like the dup. code. 
I would go for `if (_position < _length && Peek() == '}')`.
This also removes the local and makes the braces around `case` block unrequired (unless that's your coding style).